BLSQ/iaso
# WC2-217: add param to get form descriptor

Needed to get form descriptor from instances API (list), and make payload more consistent between list and detail

Related JIRA tickets : IA-2140, WC2-217

## Self proofreading checklist


## Changes

- Make  `Instance`s  `as_full_model` return `created_by` and make `as_dict` return `last_modified_by`. This will make the UI more consistent. 
- Add `as_dict_with_descriptor` which is then used when the front end passes the `with_descriptor` query param

## How to test

If you're set up for deduplication (I'll post the doc when it's available, otherwise just ping me):
- Go to the details of a duplicate and expland one of its submissions. you should see the labels of the questions, not just the keys
- Also, the Information part should have values for both "Created by" and " Last modified by"

If you're not set up:
- Go to submission page and search for a specific form (to avoid making to the call too heavy)
- In the codebase, find the api call and add the query param: `with_descriptor=true`
- In the inspector or in Django's browsable api page, check the payload of the api call to instances, it should include the form descriptor for each instance

In any case:
- Go to the details page of any instance: you should see values for both created by and last modified by


## Print screen / video

- Instance details has both created by and last modified by fields:
![Screenshot 2023-05-05 at 15 08 35](https://user-images.githubusercontent.com/38907762/236466182-6facab13-4dab-486a-937c-256f9f0462a6.png)

- Deduplication details before:
![Screenshot 2023-05-05 at 15 08 35](https://user-images.githubusercontent.com/38907762/236466362-46c3964d-e411-4698-82ac-0ea6c3d9ef3a.png)

- Deduplication details after:

![Screenshot 2023-05-05 at 15 07 52](https://user-images.githubusercontent.com/38907762/236466387-9187d747-c762-4235-8994-ed3eb21ff952.png)

## Notes
Merges into a the dev branch for deduplication
